### PR TITLE
Fix Canonical URL updater form

### PR DESCRIPTION
**Pull request by hvelarde**
A new upgrade step is also provided to update the `objects_provides` catalog index.

WIP

fixes: #115

### CODE REVIEW

File: sc/social/like/tests/test_upgrades.py [hunk 1]
```diff
@@ -285,3 +286,39 @@ def test_enable_social_media_behavior(self):
...
+
+
+class To3046TestCase(UpgradeTestCaseBase):
+
+    def setUp(self):
+        UpgradeTestCaseBase.setUp(self, u'3045', u'3046')
+
+    def test_upgrade_to_3046_registrations(self):
+        version = self.setup.getLastVersionForProfile(self.profile_id)[0]
+        self.assertGreaterEqual(int(version), int(self.to_version))
```
Comment (hvelarde) [inline sc/social/like/tests/test_upgrades.py]: @tcurvelo I have no idea why this is failing, I think is something from another test; do you mind to dig on it?

Comment (hvelarde) [inline sc/social/like/tests/test_upgrades.py]: I already found the issue and fixed it: https://github.com/collective/sc.social.like/pull/118/commits/78bc1e17e1e99c7a4c833a3d1c2600bb232321f3

File: sc/social/like/browser/canonicalurl.py [hunk 2]
```diff
@@ -42,11 +30,12 @@ class ICanonicalURLUpdater(model.Schema):
         constraint=validate_canonical_domain,
     )
 
-    created_before = schema.Date(
+    published_before = schema.Date(
```
Comment (rodfersou) [inline sc/social/like/browser/canonicalurl.py]: I miss the 'date' name in this attribute, how about to change it to 'published_before_date'?

File: sc/social/like/tests/test_canonicalurl_view.py [hunk 3]
```diff
@@ -23,21 +24,31 @@ def setUp(self):
...
+            obj = api.content.create(self.portal, type='News Item', id='baz')
+            api.content.transition(obj, 'publish')
 
         # simulate objects were create way long in the past
-        self.portal['foo'].creation_date = DateTime('2015/01/01')
-        self.portal['bar'].creation_date = DateTime('2016/01/01')
+        self.portal['foo'].effective_date = DateTime('2015/01/01')
         self.portal['foo'].reindexObject()
+        self.portal['bar'].effective_date = DateTime('2016/01/01')
         self.portal['bar'].reindexObject()
```
Comment (rodfersou) [inline sc/social/like/tests/test_canonicalurl_view.py]: we can add the date in the `api.content.create` call avoiding the need to reindex the object

**Comment by hvelarde** [comment]
I opened this thread on the community forum to find a solution to the error in Plone 5.1:

https://community.plone.org/t/strange-catalog-behavior-on-plone-5-1/4582

File: sc/social/like/tests/test_canonicalurl_view.py [hunk 4]
```diff
@@ -23,21 +24,31 @@ def setUp(self):
         self.portal = self.layer['portal']
         self.request = self.layer['request']
         enable_social_media_behavior()
+        self.setup_content()
+        api.portal.set_registry_record(
+            name='canonical_domain', value='https://example.org', interface=ISocialLikeSettings)
```
Comment (rodfersou) [inline sc/social/like/tests/test_canonicalurl_view.py]: I thought `canonical_domain` becomes `old_canonical_domain`

Comment (hvelarde) [inline sc/social/like/tests/test_canonicalurl_view.py]: only in the form.

**Pull request merged**